src/occupational_classification_utils/embed/embedding.py

```diff
@@ -0,0 +1,295 @@
+"""This module provides utilities for embedding and searching occupational classification data
+using Chroma vector stores and language models.
+
+It includes functionality for embedding SOC hierarchy data, managing vector stores,
+and performing similarity searches.
+"""
+
+import logging
+import uuid
+from typing import Any, Optional, Union
+
+from autocorrect import Speller
+from langchain.docstore.document import Document
+from langchain_community.embeddings import HuggingFaceEmbeddings, VertexAIEmbeddings
+from langchain_community.vectorstores import Chroma  # pylint: disable=no-name-in-module
+from occupational_classification.hierarchy.soc_hierarchy import SOC
+
+from occupational_classification_utils.utils.soc_data_access import (
+    load_soc_index,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# Share configuration with other modules
+embedding_config = {
+    "embedding_model_name": "unknown",
+    "llm_model_name": "unknown",
+    "db_dir": "unknown",
+    "soc_index": "unknown",
+    "soc_structure": "unknown",
+    "soc_condensed": "unknown",
+    "matches": 0,
+    "index_size": 0,
+}
+
+
+def get_config() -> dict[str, dict[str, str]]:
+    """Returns the configuration dictionary for the LLM.
+
+    Returns:
+        dict: A dictionary containing configuration details for the embedding model
+        and lookup file paths.
+    """
+    return {
+        "llm": {
+            "llm_model_name": "gemini-1.5-flash",
+            "embedding_model_name": "all-MiniLM-L6-v2",
+            "db_dir": "src/occupational_classification_utils/data/vector_store",
```

Comment:
I updated sic-classification-utils to use package references rather than file paths.  This is fine for now, but when we want to get this running as a standalone vector store it will require updating (just a heads up).

Reply:
Good point, will look at sic-classification-utils implementation.